addons/sale_stock/security/sale_stock_security.xml

```diff
@@ -10,5 +10,13 @@
             <field name="name">Display incoterms on Sales Order and related invoices</field>
             <field name="category_id" ref="base.module_category_hidden"/>
         </record>
+
+        <!-- Stock Portal Access Rules -->
+        <record id="stock_picking_rule_portal" model="ir.rule">
+            <field name="name">Portal Follower Pickings</field>
+            <field name="model_id" ref="stock.model_stock_picking"/>
+            <field name="domain_force">['|', '|', ('message_partner_ids', 'in', [user.partner_id.id]), ('partner_id', '=', user.partner_id), ('sale_id.partner_id', '=', user.partner_id)]</field>
```

Comment:
Left of domain must be without .id, but right of domain not must be recordset, only pass id of record